Add an option to disable multicast route per-network

bboreham opened the pull request:
This gives users a better way out of the problem in #1960

rade commented:
Docs?

Does docker allow you to specify option *flags*, i.e. no value, e.g. `--opt works.weave.nomulticast`? That would be marginally preferable in the UI, though I gather that the option actually acts as an override for whatever the overall plugin setting ("--no-multicast" or not) is, and for that the explicit true/false does work better.

bboreham commented:
Yes, you can say `docker network create --driver=weavemesh --opt works.weave.nomulticast weave2`.  What arrives at the plugin is a map from `works.weave.nomulticast` to blank.

bboreham commented:
I amended the code to look for `nomulticast`, and added docs.  Ping @abuehrle for docs review.

Note that you can say `--opt works.weave.nomulticast=false` if you want to invert the default.  I didn't document this, as it seems a niche interest.

rade commented:
See [my comment](https://github.com/weaveworks/weave/issues/1960#issuecomment-222823591) how I reckon the various options should interplay. In summary

- get rid of the `--no-multicast-route` option on the plugin
- created networks have multicast disabled by default
- supplying the `works.weave.multicast` option on network creation enables multicast
- create the `weave` network with multicast enabled

bboreham commented:
> get rid of the `--no-multicast-route` option on the plugin

assume you mean "issue a deprecation warning and ignore"

bboreham commented:
Changed as suggested

rade commented on prog/plugin/main.go:
```diff
@@ -73,26 +73,29 @@ func main() {
 	Log.Println("Weave plugin", version, "Command line options:", os.Args[1:])
+	if noMulticastRoute {
+		Log.Warning("--no-multicast-route option has been removed; multicast is off by default")
+	}
```
we may want to provide a clue as to how to enable it.

bboreham replied:
maybe, but presumably a person who specified `--no-multicast-route` is not the target audience.

rade replied:
d'oh, right.

rade commented on site/plugin/plugin-how-it-works.md:
```diff
@@ -26,6 +26,13 @@ store, but the first part of
+   route for multicast traffic onto its interface.
+
+>**Note:** If you connect a container to multiple Weave networks, at
+   most one of them can have the multicast route enabled.
```
we ought to mention somewhere that the `weave` network has multicast enabled.

awh commented:
LGTM

bboreham commented:
Added mention of how `--net=weave` has multicast turned on but `docker network create --driver=weave` will default to no multicast.  Also removed obsolete flag from docs.
